components/python/dbus-python/Makefile

```diff
@@ -33,20 +35,71 @@ COMPONENT_LICENSE_FILE=	COPYING
 
 PATH=$(PATH.gnu)
 
-PYTHON_VERSION = 3.5
-PYTHON_VERSIONS = 3.5
+PYTHON_VERSIONS = 3.5 3.7 3.9
 
 include $(WS_MAKE_RULES)/common.mk
 
-#COMPONENT_PREP_ACTION = (cd $(@D) && autoreconf -fi)
+# Rename SO files to cpython names
+CPIA.3.5 = ( cd $(PROTO_DIR)/$(PYTHON_VENDOR_PACKAGES) && \
+	$(MV) _dbus_bindings.so _dbus_bindings.cpython-35m.so && \
+	$(MV) _dbus_glib_bindings.so _dbus_glib_bindings.cpython-35m.so ) ;
+CPIA.3.7 = ( cd $(PROTO_DIR)/$(PYTHON_VENDOR_PACKAGES) && \
+	$(MV) _dbus_bindings.so _dbus_bindings.cpython-37m.so && \
+	$(MV) _dbus_glib_bindings.so _dbus_glib_bindings.cpython-37m.so ) ;
+CPIA.3.9 = ( cd $(PROTO_DIR)/$(PYTHON_VENDOR_PACKAGES) && \
+	$(MV) _dbus_bindings.so _dbus_bindings.cpython-39.so && \
+	$(MV) _dbus_glib_bindings.so _dbus_glib_bindings.cpython-39.so ) ;
+COMPONENT_POST_INSTALL_ACTION += $(CPIA.$(PYTHON_VERSION))
 
-CONFIGURE_OPTIONS+=	--sysconfdir=/etc
-CONFIGURE_OPTIONS+=	--disable-static
-CONFIGURE_ENV+=		PYTHON=$(PYTHON)
-CONFIGURE_ENV+=		am_cv_python_pythondir="$(PYTHON.3.5.VENDOR_PACKAGES)"
-CONFIGURE_ENV+=		am_cv_python_pyexecdir="$(PYTHON.3.5.VENDOR_PACKAGES)"
+PREFIX=		     build/amd64-$(PYTHON_VERSION)/temp.solaris-2.11-i86pc.64bit-$(PYTHON_VERSION)/prefix
 
+# Copy some uninstalled files to make way for symlinks
+COMPONENT_POST_INSTALL_ACTION += \
+	$(MKDIR) $(PROTOUSRINCDIR)/dbus-1.0/dbus; \
+	$(CP) $(PREFIX)/include/dbus-1.0/dbus/dbus-python.h \
+	$(PROTOUSRINCDIR)/dbus-1.0/dbus/dbus-python.h-$(PYTHON_VERSION) ;
+COMPONENT_POST_INSTALL_ACTION += \
+	$(MKDIR) $(PROTOUSRLIBDIR)/pkgconfig; \
+	$(MKDIR) $(PROTOUSRLIBDIR)/$(MACH64)/pkgconfig; \
+	$(GSED) -e 's|^prefix=.*$$|prefix=$(USRDIR)|' \
+	$(PREFIX)/lib/pkgconfig/dbus-python.pc \
+	> $(PROTOUSRLIBDIR)/pkgconfig/dbus-python.pc-$(PYTHON_VERSION) ;
+
+COMPONENT_TEST_ENV=	PYTHONPATH=$(PROTO_DIR)$(PYTHON_VENDOR_PACKAGES)
+
+# Tests require packages: pytest py pluggy benchmark pyparsing
+COMPONENT_TEST_DIR=	$(COMPONENT_SRC)/test
+COMPONENT_TEST_CMD=     /usr/bin/pytest-$(PYTHON_VERSION)
+COMPONENT_TEST_ARGS=	test-*.py
+
+# Disable test files that have import failures or produce only errors.
+COMPONENT_PRE_TEST_ACTION = \
+( cd $(COMPONENT_TEST_DIR); \
+  for F in test-unusable-main-loop.py test-client.py test-p2p.py \
+	test-signals.py test-exception-py2.py; \
+  do \
+    test -f $$F && $(MV) $$F $$F-not; \
+    true; \
+  done )
+
+# Typical results for test target:
+#============================= test session starts ===========================
+#...
+#=============================== warnings summary ============================
+#...
+#======================== 38 passed, 4 warnings in 0.27s =====================
+
+# Added run-time dependencies
 REQUIRED_PACKAGES += runtime/python-35
+REQUIRED_PACKAGES += runtime/python-37
+REQUIRED_PACKAGES += runtime/python-39
+
+# Added test dependencies for all python versions
+REQUIRED_PACKAGES += library/python/pytest
+REQUIRED_PACKAGES += library/python/py
+REQUIRED_PACKAGES += library/python/pluggy
```

Comment:
You don't need to add py or pluggy to the test dependencies as they are runtime dependencies of pytest.
(I have learned and fixed that recently.)

Reply:
Probably we should introduce TEST_REQUIRED_PACKAGES that install only deps if tests are invoked. However this is not going to work for all packages as some of them build tests only if the testing framework is detected.

Reply:
I've removed those two dependencies.